airflow/providers/amazon/aws/sensors/batch.py

```diff
@@ -115,7 +115,15 @@ def execute_complete(self, context: Context, event: dict[str, Any]) -> None:
         Relies on trigger to throw an exception, otherwise it assumes execution was successful.
         """
         if event["status"] != "success":
-            raise AirflowException(f"Error while running job: {event}")
+            message = f"Error while running job: {event}"
+            # TODO: remove this if-else block when min_airflow_version is set to higher than the version that
+            # raise_failed_or_skiping_exception is introduced in BaseSensorOperator
+            if getattr(self, "raise_failed_or_skiping_exception"):
+                self.raise_failed_or_skiping_exception(failed_message=message)
```

Comment:
I will open a discussion on slack, related to this, could we wait before merging?